### PR TITLE
feat(Library):adding-tooltip-for-node-category

### PR DESCRIPTION
**Pull request by Enzo707**
tooltips

### Purpose

There are three categories of nodes inside Dynamo; Creators, Actions and Queries, which are currently demarcated by a green cross, orange lighting bolt and blue question mark. This PR add tooltips to make it easier to explaining what each of these buckets of nodes does.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

Now a tooltip will shown over each node category displaying a brief explanation about what does the node bucket do.

### Reviewers

@QilongTang

### CODE REVIEW

File: src/LibraryViewExtensionWebView2/web/library/resources/library-action-old.svg
Comment (QilongTang) [inline src/LibraryViewExtensionWebView2/web/library/resources/library-action-old.svg:1]: hmm what are these changes about new icons?

File: src/LibraryViewExtensionWebView2/web/library/library.html [hunk 1]
```diff
@@ -105,9 +105,9 @@
         let intervalId;
 
         // Disable the context menu
-        document.oncontextmenu = function () {
-            return false;
-        }
+       // document.oncontextmenu = function () {
+         //   return false;
+        //}
```
Comment (QilongTang) [inline src/LibraryViewExtensionWebView2/web/library/library.html:110]: Do we still need these lines?

File: src/PackageDetailsViewExtension/PackageDetailsViewExtension.csproj [hunk 2]
```diff
@@ -48,6 +48,7 @@
       <Name>DynamoServices</Name>
       <Private>False</Private>
     </ProjectReference>
+    <ProjectReference Include="..\PythonMigrationViewExtension\PythonMigrationViewExtension.csproj" />
```
Comment (QilongTang) [inline src/PackageDetailsViewExtension/PackageDetailsViewExtension.csproj:51]: What is the purpose of this new dependency?

File: src/LibraryViewExtensionWebView2/web/library/resources/library-action-old.svg
Comment (Enzo707) [reply src/LibraryViewExtensionWebView2/web/library/resources/library-action-old.svg:1]: Let me check that...

File: src/PackageDetailsViewExtension/PackageDetailsViewExtension.csproj (hunk 2 above)
Comment (Enzo707) [reply src/PackageDetailsViewExtension/PackageDetailsViewExtension.csproj:51]: Not sure why this change was added. I didn't change this at least manually. I'm gonna removed that

File: src/LibraryViewExtensionWebView2/LibraryViewController.cs [hunk 3]
```diff
@@ -24,6 +25,7 @@
 using Microsoft.Web.WebView2.Core;
 using Microsoft.Web.WebView2.Wpf;
 using Newtonsoft.Json;
+using ProtoCore.Lang;
```
Comment (QilongTang) [inline src/LibraryViewExtensionWebView2/LibraryViewController.cs:28]: Are these new using valid?

File: src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs [hunk 4]
```diff
@@ -64,4 +65,4 @@
 // You can specify all the values or you can default the Build and Revision Numbers
 // by using the '*' as shown below:
 // [assembly: AssemblyVersion("1.0.*")]
-[assembly: AssemblyFileVersion("2.18.0.2986")]
+[assembly: AssemblyFileVersion("2.18.0.4447")]
```
Comment (QilongTang) [inline src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs:68]: We usually skip the changes to this file, would you revert the changes to this file?

File: src/LibraryViewExtensionWebView2/web/library/librarie.min.js
Comment (QilongTang) [inline src/LibraryViewExtensionWebView2/web/library/librarie.min.js:1]: I assume this is the updated minified version based on changes in the other repo

File: src/LibraryViewExtensionWebView2/web/library/resources/library-arrow-right-centered.svg
Comment (QilongTang) [inline src/LibraryViewExtensionWebView2/web/library/resources/library-arrow-right-centered.svg:1]: What is the use of this additional here, I do not see this got referenced anywhere

**Review by QilongTang — COMMENTED** [review]
Last couple of comments

**Comment by QilongTang** [comment]
Only sporadic test failure reported from run https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/8776/testReport/. Will merge once basic checks passed, these are triggered because I removed one empty line

File: src/Libraries/CoreNodeModels/Properties/Resources.Designer.cs [hunk 5]
```diff
@@ -19,7 +19,7 @@ namespace CoreNodeModels.Properties {
     // class via a tool like ResGen or Visual Studio.
     // To add or remove a member, edit your .ResX file then rerun ResGen
     // with the /str option, or rebuild your VS project.
-    [global::System.CodeDom.Compiler.GeneratedCodeAttribute("System.Resources.Tools.StronglyTypedResourceBuilder", "16.0.0.0")]
```
[the remark this answers is not in the document]
Comment (QilongTang) [reply src/Libraries/CoreNodeModels/Properties/Resources.Designer.cs:22]: hi @RobertGlobant20 I think v16 belongs to VS2019 and v17 belongs to VS2022. In the past, when we have multiple devs using different versions of VS, there is always this version change back and forth. But I would not expect this impacting compilation result. Do you start to see failure on master now? It builds fine for me on VS2022

[the remark this answers is not in the document]
Comment (RobertGlobant20) [reply src/Libraries/CoreNodeModels/Properties/Resources.Designer.cs:22]: There are no failures, then if it builds I think we are fine.
